draft-duke-quic-load-balancers.md

```diff
@@ -59,270 +61,545 @@ author:
+   into the remaining required connection ID octets. These arbitrary
+   bits MAY encode additional information, but SHOULD appear
+   essentially random to observers.
+
+   The server then encrypts the server ID bytes using 128-bit AES in
+   counter (CTR) mode, much like QUIC packet number decryption. The counter
+   input to AES-CTR is the bytes of the connection ID that do not
+   constitute the encrypted server ID.
+
+   encrypted_server_id = AES-CTR(key, non_server_id_bytes, server-id)
+   
+# Protocol Description {#protocol-description}
+
+   The fundamental protocol requirement is to share the choice of
+   routing algorithm, and the relevant parameters for that algorithm,
+   between load balancer and server.
+
+   For Plaintext CID Routing, this consists of the Routing Bits,
+   Divisor, and Modulus. The Modulus is unique to each server,
+   but the others MUST be global.
+
+   For Encrypted CID Routing, this consists of the Server ID,
+   Server ID Length, Key, and Connection ID Length. The Server ID
+   is unique to each server, but the others MUST be global.
+
+## Out of band sharing
+
+   When there are concerns about the integrity of the path between
+   load balancer and server, operators may share routing information
+   using an out-of-band technique, which is out of the scope of
+   this specification.
+
+   To simplify configuration, the global parameters can be shared
+   out-of-band, while the load balancer sends the unique server
+   IDs via the truncated message formats presented below.
+
+## QUIC-LB Message Exchange
+
+   QUIC-LB load balancers send the encoding parameters to servers
+   as they discover the servers, using a single packet to each that
+   resembles QUIC. They periodically retransmit this packet to each
+   server until that server responds with a QUIC-LB ack. Specifics
+   of this retransmission are implementation-dependent.
+
+   These message formats are specific to QUICv2 and experimental
+   versions leading up to QUICv2. They may require revision for
+   future versions of QUIC.
+
+## Packet Header Format
+   QUIC-LB messages resemble QUIC packets with a long header. They are
+   sent when a load balancer boots up, or detects a new server in the
+   pool. QUIC-LB packets are delivered in a UDP datagram.
+ 0                   1                   2                   3
+ 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1 2 3 4 5 6 7 8 9 0 1
++-+-+-+-+-+-+-+-+
+| Type = 0xfb   |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                      	Version (32)                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|      0x00     |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                           Length (i)                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|     Random    |
++-+-+-+-+-+-+-+-+
+{: #quic-lb-header title="QUIC-LB Header"}
```

Comment:
I still think that this entire packet should be encrypted with a key that is configured out of band. The key could still default to a null-key, but that is implementation/deployment specific. Then, the last 16 bytes would consist of the AEAD tag, just like normal QUIC packets.

Reply:
That is a long argument that I am happy to have in future iterations, with more stakeholders present.